threshsign/src/ThresholdSignaturesTypes.cpp

```diff
@@ -21,26 +21,29 @@
 Cryptosystem::Cryptosystem(const std::string& sysType,
                            const std::string& sysSubtype,
                            uint16_t sysNumSigners,
-                           uint16_t sysThreshold) {
+                           uint16_t sysThreshold) :
+                             type(sysType),
+                             subtype(sysSubtype),
+                             numSigners(sysNumSigners),
+                             threshold(sysThreshold),
+                             forceMultisig_((numSigners == threshold)?true:false),
+                             signerID(NID),
+                             publicKey("uninitialized")
+{
   if (!isValidCryptosystemSelection(sysType, sysSubtype, sysNumSigners,
         sysThreshold)) {
     throw InvalidCryptosystemException("Invalid cryptosystem selection:"
       " primary type: " + sysType + ", subtype: " + sysSubtype + ", with "
       + std::to_string(sysNumSigners) + " signers and threshold of "
       + std::to_string(sysThreshold) + ".");
   }
-  type = sysType;
-  subtype = sysSubtype;
-  numSigners = sysNumSigners;
-  threshold = sysThreshold;
-  signerID = NID;
 }
 
 // Helper function to generateNewPseudorandomKeys.
 IThresholdFactory* Cryptosystem::createThresholdFactory() {
-  if (type == MULTISIG_BLS_SCHEME) {
+  if (type == MULTISIG_BLS_SCHEME || forceMultisig_) {
```

Comment:
Actually I preserved the behaviour, since the decision to use multisig in case of numsigners==threshold, which is forceMultisig_, was made in BlsThresholdFactory.

Reply:
Ah, I see. Thank you for pointing that out.